l10n/po/zh_CN/_versions/main/guides/kafka.adoc.po

```diff
@@ -1900,53 +1900,53 @@ msgstr ""
-msgstr ""
+msgstr "如果处理成功完成，则刷新生产者并提交事务。"
 
 #. type: Plain text
 #: upstream/_versions/main/guides/kafka.adoc:1093
 msgid "If the processing throws an exception, returns a failing `Uni`, or marks the `TransactionalEmitter` for abort, the transaction is aborted."
-msgstr ""
+msgstr "如果处理过程抛出异常，会返回失败的 `Uni`，或将 `TransactionalEmitter` 标记为中止，然后事务会被中止。"
 
 #. type: Plain text
 #: upstream/_versions/main/guides/kafka.adoc:1097
 msgid "Kafka transactional producers require configuring `acks=all` client property, and a unique id for `transactional.id`, which implies `enable.idempotence=true`.  When Quarkus detects the use of `KafkaTransactions` for an outgoing channel it configures these properties on the channel, providing a default value of `\"${quarkus.application.name}-${channelName}\"` for `transactional.id` property."
-msgstr ""
+msgstr "Kafka 事务生产者需要配置 `acks=all` 客户端属性，以及 `transactional.id` 的唯一 id，这意味着 `enable.idempotence=true` 。 当 Quarkus 检测到传出通道使用 `KafkaTransactions` 时，它会在通道上配置这些属性，为 `transactional.id` 属性提供默认值 `\"${quarkus.application.name}-${channelName}\"` 。"
 
 #. type: Plain text
 #: upstream/_versions/main/guides/kafka.adoc:1099
 msgid "Note that for production use the `transactional.id` must be unique across all application instances."
-msgstr ""
+msgstr "请注意，要在生产环境使用，必须确保 `transactional.id` 在所有应用实例中是唯一的。"
 
 #. type: delimited block =
 #: upstream/_versions/main/guides/kafka.adoc:1107
 msgid "While a normal message emitter would support concurrent calls to `send` methods and consequently queues outgoing messages to be written to Kafka, a `KafkaTransactions` emitter only supports one transaction at a time.  A transaction is considered in progress from the call to the `withTransaction` until the returned `Uni` results in success or failure.  While a transaction is in progress, subsequent calls to the `withTransaction`, including nested ones inside the given function, will throw `IllegalStateException`."
-msgstr ""
+msgstr "虽然普通的消息emitter支持并发调用 `send` 方法并将要写入 Kafka 的传出消息顺序排队，但 `KafkaTransactions` emitter每次只支持一个事务。 从调用 `withTransaction` 直到返回的 `Uni` 导致成功或失败，事务被视为正在进行中。 当事务正在进行时，对 `withTransaction` 的后续调用，包括给定函数内的嵌套调用，都将抛出 `IllegalStateException` 。"
 
 #. type: delimited block =
 #: upstream/_versions/main/guides/kafka.adoc:1111
 msgid "Note that in Reactive Messaging, the execution of processing methods, is already serialized, unless `@Blocking(ordered = false)` is used.  If `withTransaction` can be called concurrently, for example from a REST endpoint, it is recommended to limit the concurrency of the execution.  This can be done using the `@Bulkhead` annotation from link:https://quarkus.io/guides/smallrye-fault-tolerance[_Microprofile Fault Tolerance_]."
-msgstr ""
+msgstr "请注意，在 Reactive Messaging 中，处理方法的执行已经被序列化，除非使用了 `@Blocking(ordered = false)` 。 如果可以同时调用 `withTransaction` ，例如从 REST 端点，建议限制执行的并发性。 这可以使用后面链接中的 `@Bulkhead` 注释来完成：link:https://quarkus.io/guides/smallrye-fault-tolerance[_Microprofile Fault Tolerance_]。"
```

Comment:
I'm not sure if there is best practice for "endpoint" translation. From my personal view, I'd prefer to use "节点" instead of "端点" for it. But if there is best practice I'm ok with it.

Reply:
Done